tests/multi_ptr/multi_ptr_common.h

```diff
@@ -163,6 +163,26 @@ struct check_pointer {
   }
 };
 
+/** @brief Initializes all elements in a container with 0
+ */
+template <typename T, int N, int... Vals>
+struct zero_init_container : zero_init_container<T, N - 1, 0, Vals...> {};
+template <typename T, int... Vals>
+struct zero_init_container<T, 0, Vals...> {
+  using elem_type = typename T::value_type;
+  static inline constexpr T value = {elem_type{Vals}...};
+};
+
+/** @brief Initializes an std::array with either 0's or using the default
```

Comment:
I do not understand the relation between the comment and the following.

Reply:
`user_def_types::def_cnstr` is the only class in use that does not have a constructor taking an integer, so we make an exception to the 0-initialization. We could also do it the other way around, i.e. only do explicit initialization of `no_def_cnstr` given it is the only one without a default constructor.